### ISSUE TITLE
Lint: Disallow 0-tuples and 1-tuples

### ISSUE THREAD

**Issue by TanklesXL** [bcpeinhardt/code_review#25]
**What the rule should do:**

Warn on uses of 0-tuples and 1-tuples

Fix:
- rewrite 0-tuples to use `Nil`
- rewrite 1-tuples to use the contained value directly
 

**What problems does it solve:**
0 and 1 arity tuples could be common in the language that a user is more familiar with but are not idiomatic in Gleam. This lint helps to steer code into a more idiomatic and common form.

**Example of things the rule would report:**

```gleam
// 0-tuple
pub fn something_with_a_side_effect() -> #() {
...
}

// 1-tuple
pub fn map_tuple_1(a: #(a), f: fn(a) -> b) -> #(b) {
...
}

```

**Example of things the rule would not report:**
Nothing comes to mind at the moment.

**When (not) to enable this rule:**

0-tuple: I don't think there should be any exceptions to the rule.
1-tuple: The only thing i can think of here is when decoding `Dynamic` data but even then we can encourage the user to use the `dynamic.element` function to decode to a single value.

**I am looking for:**
- Feedback

**Comment by jfmengels** [comment]
This entire project was just a nerdsnipe to get @lpil to add more things into the compiler.

**Comment by lpil** [comment]
I don't think this should be in the compiler as I can imagine valid uses with FFI and database clients.

**Comment by inoas** [comment]
I don't think it should be in the compiler because of erlang interop where this could easily happen?
